multimodal/src/autogluon/multimodal/models/huggingface_text.py

```diff
@@ -61,18 +72,37 @@ def __init__(
                     - 'xlm-roberta-base'
         num_classes
             The number of classes. 1 for a regression task.
+        pooling_mode
+            The pooling mode for the Transformer. Can be "cls", or "mean"
+        gradient_checkpointing
+            Whether to enable gradient checkpointing
         """
         super().__init__()
         logger.debug(f"initializing {checkpoint_name}")
         self.checkpoint_name = checkpoint_name
         self.num_classes = num_classes
+
         self.model = AutoModel.from_pretrained(checkpoint_name)
+        if isinstance(self.model, T5PreTrainedModel):
+            self.is_t5 = True
+            # Remove the decoder in T5
+            del self.model.decoder
+        else:
+            self.is_t5 = False
```

Comment:
Could we wrap this into a util function to detect special models? I guess `t5` is just one of them we need to deal with specially. The return of this util function is the model type.

Reply:
Yes, but we may do it later. Encoder-decoder models like  BART should also be specially treated.

Reply:
In fact, we may need to figure out the way for supporting applying the decoder in model training.

Reply:
Maybe we need a config `uses_decoder`.

Reply:
I find that people will tend to follow previous design to add new stuff. By following this design, people will add more if-else patterns here.